proto/buf/alpha/registry/v1alpha1/repository.proto

```diff
@@ -162,6 +162,16 @@ service RepositoryService {
   rpc GetRepositoriesMetadata(GetRepositoriesMetadataRequest) returns (GetRepositoriesMetadataResponse) {
     option idempotency_level = NO_SIDE_EFFECTS;
   }
+  // GetRepositoryDefaultBranch retrieves the default_branch value for a given repository. By
+  // default a BSR repository's default branch is "main", which is the value used when a module is
+  // pushed to the BSR.
+  rpc GetRepositoryDefaultBranch(GetRepositoryDefaultBranchRequest) returns (GetRepositoryDefaultBranchResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
```

Comment:
Instead of this RPC we could include the `default_branch` as a field in the `message Repository`, but then in the backend we'd need to make sure we always include the field in all responses. Thoughts? @doriable @seankimdev

Reply:
Isn't this a duplicate rpc to the existing [GetCurrentDefaultBranch](https://buf.build/bufbuild/buf/docs/main:buf.alpha.registry.v1alpha1#buf.alpha.registry.v1alpha1.RepositoryBranchService.GetCurrentDefaultBranch)?

Reply:
I am a fan of having `default_branch` as a field on the `Repository` message. I think it provides a lot more clarity.

Also, just noting that we have moved the discussion on whether this encroaches on `GetCurrentDefaultBranch` onto Slack.